### PR TITLE
Exception handling for ProtocolError in operator

### PR DESCRIPTION
**Pull request by leelavg**
Updates: #585

Signed-off-by: Leela Venkaiah G <leelavg@thoughtexpo.com>

### CODE REVIEW

File: operator/main.py [hunk 1]
```diff
@@ -838,28 +839,38 @@ def crd_watch(core_v1_client, k8s_client):
...
+                    handle_modified(core_v1_client, obj)
+                elif operation == "DELETED":
+                    handle_deleted(core_v1_client, obj)
+        except ProtocolError:
+            # It might so happen that this'll be logged for every hit in k8s
+            # stream and better to log at debug level
+            logging.debug(
+                logf(
+                    "Watch connection broken and restarting watch on the stream"
+                ))
```
Comment (aravindavk) [inline operator/main.py:880]: Add little sleep(`time.sleep(1)`) here to avoid infinite loop consuming full cpu.

File: operator/main.py [hunk 2]
```diff
@@ -838,28 +839,38 @@ def crd_watch(core_v1_client, k8s_client):
...
+                metadata = obj.get("metadata")
+                resource_version = metadata['resourceVersion']
+                logging.debug(
+                    logf("Event", operation=operation, object=repr(obj)))
+                if operation == "ADDED":
+                    handle_added(core_v1_client, obj)
+                elif operation == "MODIFIED":
+                    handle_modified(core_v1_client, obj)
+                elif operation == "DELETED":
+                    handle_deleted(core_v1_client, obj)
```
Comment (aravindavk) [inline operator/main.py]: May be good to make this a function?

File: operator/main.py (hunk 1 above)
Comment (leelavg) [reply operator/main.py:880]: sure, do you think adding a threshold counter for number of tries would also be good? Like raise exception after ~10 times hitting the exception and the operator will be restarted?

File: operator/main.py (hunk 2 above)
Comment (leelavg) [reply operator/main.py]: ack.

File: operator/main.py (hunk 1 above)
Comment (aravindavk) [reply operator/main.py:880]: Restarting may not help if the problem persists. Increase the sleep duration to 30 seconds?

Comment (leelavg) [reply operator/main.py:880]: ack

**Review by amarts — APPROVED** [review]

**Review by aravindavk — APPROVED** [review]

**Pull request merged**